src/types/cart.ts

```diff
@@ -0,0 +1,73 @@
+import { z } from "zod";
+
+const addressSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+  address1: z.string(),
+  address2: z.string().nullable(),
+  address3: z.string().nullable(),
+  city: z.string(),
+  state: z.string(),
+  subdivision_code: z.string().nullable(),
+  postal_code: z.string(),
+  country_code: z.string(),
+});
+
+const shippingOptionSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  delivery_time: z.string(),
+  price: z.number(),
+  price_label: z.string(),
+});
+
+const paymentMethodSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  icons: z.array(z.string()),
+  description: z.string(),
+});
+
+const cartItemsSchema = z.object({
```

Comment:
is the id here the product id? Maybe we could map over these and get the products kinda like collections does.